tests/st/calico_upgrade/test_calico_upgrade.py

```diff
@@ -0,0 +1,206 @@
+        rcu.assert_no_error()
+
+        ready_output = get_value_etcdv2("/calico/v1/Ready")
+        logger.debug("INFO: etcdv2 Ready flag is set to:\n%s" % ready_output)
+        assert ready_output == "true"
+
+        dry_run_output = open("/code/calico-upgrade-report/convertednames", 'rU')
+        dry_run_output_lines = dry_run_output.readlines()
+        dry_run_output.close()
+        logger.debug(
+            "INFO: calico-upgrade dry-run convertednames output:\n%s" % dry_run_output_lines)
+
+        rcu = calicoupgrade("start")
+        logger.debug("INFO: calico-upgrade start should return 0.")
+        rcu.assert_no_error()
+
+        ready_output = get_value_etcdv2("/calico/v1/Ready")
+        logger.debug("INFO: etcdv2 Ready flag is set to:\n%s" % ready_output)
+        assert ready_output == "false"
+
+        datastore_ready_rc = _get_ready_etcdv3()
+        assert datastore_ready_rc is False
+
+        start_output = open("/code/calico-upgrade-report/convertednames", 'rU')
+        start_output_lines = start_output.readlines()
+        start_output.close()
+        logger.debug("INFO: calico-upgrade start convertednames output:\n%s" % start_output_lines)
+
+        if dry_run_output_lines == start_output_lines:
+            logger.debug("INFO: calico-upgrade dry-run and calico-upgrade start output are equal")
+        else:
+            assert dry_run_output_lines == start_output_lines
+
+        rcc = calicoctl("convert", data=testdata)
+
+        rcc.assert_no_error()
+        parsed_output = yaml.safe_load(rcc.output)
+        converted_data = clean_calico_data(parsed_output)
+        logger.debug("INFO: converted data to v3\n%s" % converted_data)
+        original_resource = rcc
+
+        rcc = calicoctl("get %s %s -o yaml" % (converted_data['kind'], name(converted_data)))
+        logger.debug("INFO: calicoctl (v3) get - after calico-upgrade start: \n%s" % rcc.output)
+
+        # Comparison here needs to be against cleaned versions of data to remove Creation Timestamp
+        logger.debug("Comparing 'get'ted output with original converted yaml")
+        cleaned_output = yaml.safe_dump(
+            clean_calico_data(
+                yaml.safe_load(rcc.output),
+                extra_keys_to_remove=['projectcalico.org/orchestrator', 'namespace']
+            )
+        )
+        original_resource.assert_data(cleaned_output)
+
+        rcu = calicoupgrade("complete")
+        logger.debug("INFO: calico-upgrade complete should return 0.")
+        rcu.assert_no_error()
+
+        ready_output = get_value_etcdv2("/calico/v1/Ready")
+        logger.debug("INFO: etcdv2 Ready flag is set to:\n%s" % ready_output)
+        assert ready_output == "false"
+
+        datastore_ready_rc = _get_ready_etcdv3()
+        assert datastore_ready_rc is True
+    else:
+        rcu = calicoupgrade("dry-run")
+        logger.debug("INFO: calico-upgrade dry-run should return non-zero.")
+        rcu.assert_error(error_text)
+
+
```

Comment:
I think we are missing tests that run the `abort` command.  This command should re-enable the v1 Ready flag.

Reply:
added test_start_abort_ignore_v3_data()